### PR TITLE
293 merge reachy2sdk with mobile base sdk

### PR DESCRIPTION
**Pull request by glannuzel**
Based on #289 

No more call to the mobile base sdk, the mobile base is now a class of reachy2 sdk

### CODE REVIEW

File: src/reachy2_sdk/parts/mobile_base.py [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+        return ZuuuModePossiblities.keys()[mode_id]
+
+    def _get_control_mode(self) -> ControlModePossiblities:
+        mode_id = self._utility_stub.GetControlMode(Empty()).mode
+        return ControlModePossiblities.keys()[mode_id]
+
+    @property
+    def battery_voltage(self) -> float:
+        """Return the battery voltage. Battery should be recharged if it reaches 24.5V or below."""
+        return float(round(self._utility_stub.GetBatteryLevel(Empty()).level.value, 1))
```
Comment (FabienDanieau) [inline src/reachy2_sdk/parts/mobile_base.py:99]: We can print a warning if battery_voltage is below 24.5. What do you think?

File: src/reachy2_sdk/parts/mobile_base.py [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+        req = GoToVector(
+            x_goal=FloatValue(value=x),
+            y_goal=FloatValue(value=y),
+            theta_goal=FloatValue(value=deg2rad(theta)),
+        )
+        self._drive_mode = "go_to"
+        self._mobility_stub.SendGoTo(req)
+
+        tic = time.time()
+        arrived: bool
```
Comment (FabienDanieau) [inline src/reachy2_sdk/parts/mobile_base.py:232]: initialize arrived with a value

Comment (FabienDanieau) [reply src/reachy2_sdk/parts/mobile_base.py:232]: This while means that we are waiting for the sendgoto to complete right? Maybe we can add a comment or move it to a function so it's more explicit

File: src/reachy2_sdk/subparts/lidar.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+
+class Lidar:
+    """LIDAR class for mobile base SDK."""
+
+    def __init__(self, initial_state: LidarObstacleDetectionStatus, grpc_channel: grpc.Channel) -> None:
+        """Initialize the LIDAR class."""
+        self._logger = logging.getLogger(__name__)
+        self._stub = MobileBaseLidarServiceStub(grpc_channel)
+
+        self._safety_enabled: bool
```
Comment (FabienDanieau) [inline src/reachy2_sdk/subparts/lidar.py:34]: explicitly init the default values

File: src/reachy2_sdk/subparts/lidar.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+        return self._obstacle_detection_status
+
+    def reset_safety_default_values(self) -> None:
+        """Reset default distances values for safety detection.
+
+        Reset values are:
+        - safety_critical_distance
+        - safety_slowdown_distance.
+        """
+        self.safety_critical_distance = 0.55
```
Comment (FabienDanieau) [inline src/reachy2_sdk/subparts/lidar.py:135]: See comment on init. We can set a "constant" self.SAFETY_CRITICAL_DISTANCE_DEFAULT = 0.55 and use it here and in the init

File: src/reachy2_sdk/subparts/__init__.py [hunk 5]
```diff
@@ -0,0 +1,6 @@
+"""
+# Subparts module.
```
Comment (FabienDanieau) [inline src/reachy2_sdk/subparts/__init__.py:2]: Not a big fan of the subparts name. If we keep this we should move it inside parts folder. But parts mean more a body part for us.
Why not sensors? We may have more in the future btw

**Review by FabienDanieau — CHANGES_REQUESTED** [review]
Looks good although I cannot really test with the fake robot.
We should add offline tests though as we did for the other classes

File: src/reachy2_sdk/parts/mobile_base.py (hunk 1 above)
Comment (glannuzel) [reply src/reachy2_sdk/parts/mobile_base.py:99]: Added in #294

File: src/reachy2_sdk/subparts/lidar.py (hunk 3 above)
Comment (glannuzel) [reply src/reachy2_sdk/subparts/lidar.py]: This is done in #294 I think

File: src/reachy2_sdk/subparts/lidar.py (hunk 4 above)
Comment (glannuzel) [reply src/reachy2_sdk/subparts/lidar.py]: Same, all is correctly init in #294

File: src/reachy2_sdk/parts/mobile_base.py [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+
+        req = GoToVector(
+            x_goal=FloatValue(value=x),
+            y_goal=FloatValue(value=y),
+            theta_goal=FloatValue(value=deg2rad(theta)),
+        )
+        self._drive_mode = "go_to"
+        self._mobility_stub.SendGoTo(req)
+
+        arrived = self._is_arrived_in_given_time(time.time(), timeout, tolerance)
```
Comment (FabienDanieau) [inline src/reachy2_sdk/parts/mobile_base.py:231]: a await is missing since we call a async function: arrived = await self...

**Review by FabienDanieau — APPROVED** [review]
just a minor comment but it looks good

**Pull request merged**